src/rascal_utility.hh

```diff
@@ -263,7 +267,16 @@ namespace rascal {
       static const std::string GetTypeName() {
 // The output of of Pretty Function depends on the compiler
 // the #define strings is a pain to split
-#if defined(GCC_COMPILER)
+#if defined(GCC_COMPILER_9_AND_UPPER)
+#define FUNCTION_MACRO __PRETTY_FUNCTION__
+#define PREFIX                                                                 \
+  "static const string rascal::internal::GetTypeNameHelper<T>::GetTypeName() " \
+  "[with T = "  // NOLINT
+#define SUFFIX_1                                                               \
+  "; std::string = std::__cxx11::basic_string<char>]"  // NOLINT
```

Comment:
What happens if one compiles this with `-D_GLIBCXX_USE_CXX11_ABI=0`?

Reply:
I have no clue what this flag means...

Reply:
I just compiled with gcc-9 and this flag with cmake and it works

Reply:
> I have no clue what this flag means...

GCC > 5 has a different ABI when in C++11 mode (hence the  `__cxx11` in the name), and this flag force the use of the older ABI. It is required on some (old) systems, where libstdc++ is compiled with the old ABI.

But if this works for you, it should be good.

Reply:
ok sorry for the confusion. I compile with this flag and boost has some linking error because of it. I guess I would have to recompile boost test with this flag for it to work.
Again what is the point of this flag ?

Reply:
> Why would we want to use this flag while compiling ?

This flag is required on "older" Linux distributions, where the base system compiler is older than GCC 5. This means that the standard C++ library as well as most of the code will be using the older ABI, so even when using more recent compilers on this platform, you want to be compatible and use this flag. A specific example would be CentOS 7 (the latest version of CentOS), where the base compiler is GCC 4.8.5. More modern compilers such as GCC 7 are provided, but use this flag by default to be compatible with the system.

This flags was also required for me on macOS, since the Homebrew version of Boost seems to be using it, meaning that I needed to provide it.

---

Since this is somehow niche usage, I am okay with the fix in this PR, assuming we move away from this implementation and use RTTI when possible.